relay-general/src/store/transactions/processor.rs

```diff
@@ -1,21 +1,89 @@
+use chrono::Utc;
+
 use relay_common::SpanStatus;
 
 use crate::processor::{ProcessValue, ProcessingState, Processor};
 use crate::protocol::{
-    Context, ContextInner, Event, EventType, Span, Timestamp, TransactionSource,
+    Context, ContextInner, Event, EventType, Span, Timestamp, TransactionInfo, TransactionSource,
 };
 use crate::store::regexes::TRANSACTION_NAME_NORMALIZER_REGEX;
 use crate::types::{Annotated, Meta, ProcessingAction, ProcessingResult, Remark, RemarkType};
 
+use super::TransactionNameRule;
+
 /// Rejects transactions based on required fields.
 #[derive(Default)]
-pub struct TransactionsProcessor {
+pub struct TransactionsProcessor<'r> {
     normalize_names: bool,
+    tx_name_rules: &'r [TransactionNameRule],
 }
 
-impl TransactionsProcessor {
-    pub fn new(normalize_names: bool) -> Self {
-        Self { normalize_names }
+impl<'r> TransactionsProcessor<'r> {
+    pub fn new(normalize_names: bool, tx_name_rules: &'r [TransactionNameRule]) -> Self {
+        Self {
+            normalize_names,
+            tx_name_rules,
+        }
+    }
+
+    /// Applies the rule if any found to the transaction name.
+    ///
+    /// It find the first rule matching the criteria:
+    /// - source matchining the one provided in the rule sorce, default `url`
```

Comment:
There's actually no default in the code in the method below, and that's correct. Using `url` by default is something controlled by the rule generation, and that lives in sentry and should be independent of how relay behaves.
```suggestion
    /// - source matchining the one provided in the rule sorce
```